js/core/utils/callbacks.d.ts

```diff
@@ -0,0 +1,21 @@
+type CallbackType<TArgs extends any[], TContext>
+  = ((this: TContext, ...args: TArgs) => boolean)
+  | ((this: TContext, ...args: TArgs) => void);
+
+export interface Callback<TArgs extends any[] = any[], TContext = any> {
+  add(fn: CallbackType<TArgs, TContext>): this;
+
+  remove(fn: CallbackType<TArgs, TContext>): this;
+
+  has(fn: CallbackType<TArgs, TContext>): this;
+
+  empty(): this;
+
+  fireWith(context: TContext, args: TArgs): this;
+
+  fire(...args: TArgs): this;
+
+  fired(): boolean;
+}
+
+export default function Callbacks<TArgs extends any[]>(): Callback<TArgs>;
```

Comment:
**Minor**
Let's rename it to `createCallback` (both in  .ts and in .js). This is a default export, nothing should be broken.

```suggestion
export default function createCallback<TArgs extends any[]>(): Callback<TArgs>;
```